### DISCUSSION TITLE
jspsych-survey-text plugin advances when pushing space bar

### DISCUSSION THREAD

**Discussion by BrianOdegaard** [jspsych/jsPsych#1107] [Q&A]
Hi everyone - 

Is there a way to combine using the fullscreen plugin and jspsych-survey-text so that when you press space bar, it DOESN'T advance to the next screen?

We have an experiment where we use jspsych-survey-text a lot and participants need to write out complete sentences to respond to some of the prompts, but whenever you push space bar, it executes the same action as pushing the "continue" button we made on the jspsych-survey-text page, and the experiment advances.

Just wondering if there is a quick solution to prevent space bar from advancing the experiment.
Thx,
Brian

**Comment by becky-gilbert** [comment]
Hi Brian, I checked to see if the space bar ends a survey-text trial using this code in Chrome:
```javascript
    var fullscreen_trial = {
        type: 'fullscreen',
        fullscreen_mode: true
    };

    var trial = {
        type: 'survey-text',
        questions: [{prompt: 'test', rows: 1, columns: 40}]
    };

    jsPsych.init({
        timeline: [fullscreen_trial, trial, trial],
        on_finish: function() {
            jsPsych.data.displayData();
        }
    });
```
And I wasn't able to reproduce the issue - when I run this, the space bar doesn't cause the experiment to advance to the next screen. What happens when you run this? And does the issue only happen in certain browsers?

**Reply by BrianOdegaard** [reply]
Hi Becky - 

We figured out the issue!  It was due to drawing to Canvas in a different plugin in the timeline that was being implemented.  Here was what my undergrad RA emailed to me:

"The canvas plugin was also able to advance when the space bar was pressed, and I think because I added both the canvas plugin and the survey-text plugin to the variable timeline, advancing using the space bar carried through. (I don't know if this is the technical description of what was happening, but it makes sense to me) So I added a line of code that does not allow the canvas plugin to advance through any key on the keyboard. This fixed the issues with the survey-text plugin. It should work now!"

And we tested it out and it works perfectly.  
Thanks for the help,
Brian

**Reply by jodeleeuw** [reply]
Hey Brian,

That sounds like a bug in the canvas plugin. Is this one of the canvas plugins that we are planning to add to the library? Can you see where your student got it from?
